pkg/util/proxy/proxy.go

```diff
@@ -238,3 +271,18 @@ func SkipGroup(group string) bool {
 		return false
 	}
 }
+
+func generateMd5HashForCluster(cluster *clusterapis.Cluster) string {
+	usedFields := make([]string, 0)
+	proxyHeaderKeys := make([]string, 0, len(cluster.Spec.ProxyHeader))
+	for key := range cluster.Spec.ProxyHeader {
+		proxyHeaderKeys = append(proxyHeaderKeys, key)
+	}
+	sort.Strings(proxyHeaderKeys)
+	for _, key := range proxyHeaderKeys {
+		usedFields = append(usedFields, key, cluster.Spec.ProxyHeader[key])
+	}
+	usedFields = append(usedFields, string(cluster.UID), cluster.Spec.ProxyURL, cluster.Spec.APIEndpoint)
```

Comment:
is UID used by `createProxyTransport`?

Reply:
UID changed means the cluster rejoined,   should create a new transport.